src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/decode/ValueDecoder.scala

```diff
@@ -67,21 +68,21 @@ private[decode] object ValueDecoder {
           value.toInt.some.asRight
         } catch {
           case _: NumberFormatException =>
-            (key, s"Cannot parse key $key with value $value into integer").asLeft
+            InvalidValue(key.toString, s"Cannot parse key $key with value $value into integer").asLeft
         }
     }
 
   implicit final val uuidColumnDecoder: ValueDecoder[UUID] =
     fromFunc[UUID] {
       case (key, value) =>
         if (value.isEmpty)
-          (key, s"Field $key cannot be empty").asLeft
+          InvalidValue(key.toString, s"Field $key cannot be empty").asLeft
         else
           try {
-            UUID.fromString(value).asRight[(Key, String)]
+            UUID.fromString(value).asRight[InvalidValue]
```

Comment:
`InvalidValue` is what we call "data constructor" - its better to not have it in a type position. Type is `ParsingError`, which is "root" of ADT.